File: 2.add-two-numbers.cs

using System;
using System.Collections.Generic;
using System.Linq;
 
 
public class Solution2 {
    public class ListNode 
    {
        public int val;
        public ListNode next;
        public ListNode(int x) { val = x; }
    }

    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        if (l1 == null || l2 == null)
        {
            return null;
        }

        ListNode l = null;

        ListNode n1 = l1;
        ListNode n2 = l2;
        ListNode n = l;
        int carry = 0;
        while (n1 != null || n2 != null)
        {
            var sum = carry;
            if (n1 != null )
            {
                sum += n1.val;
                n1 = n1.next;
            }
            if (n2 != null)
            {
                sum += n2.val;
                n2 = n2.next;
            }

            carry = sum/10;
            sum = sum %10;
            
            var node = new ListNode(sum);

            if (l == null)
            {
                l = node;
            }
            else
            {
                n.next = node;
            }
            n = node;

        }

        if (carry == 1)
        {
            n.next = new ListNode(1);
        }

        return l;
    }
}

File: 2_add_two_numbers.py
from typing import List, Optional

# Definition for singly-linked list.
class ListNode:
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next
         
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        result = ListNode(0)
        current = result
        while l1 or l2 or carry:
            v1 = v2 = 0
            if l1:
                v1 = l1.val
                l1 = l1.next
            if l2:
                v2 = l2.val
                l2 = l2.next
            val = v1 + v2 + carry
            if val >= 10:
                val = val -10
                carry = 1
            else:
                carry = 0
            current.next = ListNode(val)
            current = current.next
        return result.next
instance = Solution()